x/ccv/provider/keeper/proposal.go

```diff
@@ -472,28 +464,42 @@ func (k Keeper) BeginBlockCCR(ctx sdk.Context) {
 // A prop is included in the returned list if its proposed stop time has passed.
 //
 // Note: this method is split out from BeginBlockCCR to be easily unit tested.
+// TODO JEHAN: Stopping iteration here (dealt with by making a direct call to KVStorePrefixIterator)
 func (k Keeper) ConsumerRemovalPropsToExecute(ctx sdk.Context) []types.ConsumerRemovalProposal {
```

Comment:
This used to call an iterator which was stopped early. Instead, it now uses KVStorePrefixIterator directly